From 96409143bb156763798d0064b83038a2f7e7699f Mon Sep 17 00:00:00 2001
Message-ID: <96409143bb156763798d0064b83038a2f7e7699f.1742668052.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1742668052.git.sharlatanus@gmail.com>
References: <cover.1742668052.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 16 Mar 2025 23:28:54 +0000
Subject: [PATCH 14/34] gnu: python-ducc0: Update to 0.37.1.

* gnu/packages/maths.scm (python-ducc0): Update to 0.37.1.
[arguments] <test-flags>: Run tests in parallel, supported by upstream.
[native-inputs]: Remove python-wheel; add cmake-minimal,
python-nanobind, python-pytest-xdist, and python-scikit-build-core.

Change-Id: Iedc5f5ce49c50f8b72a120f592982baad06be89f
---
 gnu/packages/maths.scm | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/gnu/packages/maths.scm b/gnu/packages/maths.scm
index 8baf3a0a7f..c9c04d2abf 100644
--- a/gnu/packages/maths.scm
+++ b/gnu/packages/maths.scm
@@ -4129,7 +4129,7 @@ (define-public python-cvxopt
 (define-public python-ducc0
   (package
     (name "python-ducc0")
-    (version "0.36.0")
+    (version "0.37.1")
     (source
      (origin
        (method git-fetch)
@@ -4139,21 +4139,26 @@ (define-public python-ducc0
                       "ducc0_" (string-replace-substring version "." "_")))))
        (file-name (git-file-name name version))
        (sha256
-        (base32 "1pfj7k5s3d237r7diqrd7cgvf8p5zms6pp64nfdildx49kwggwab"))))
+        (base32 "0pckbip2ffmiwm73wrpvif3gy0a09v9b9kbyallp520l6l69n4k8"))))
     (build-system pyproject-build-system)
     (arguments
      (list
-      #:test-flags #~(list "python/test")
+      #:test-flags
+      #~(list "--numprocesses" (number->string (parallel-job-count))
+              "python/test")
       #:phases
       #~(modify-phases %standard-phases
           (add-before 'build 'set-env
             (lambda _
               (setenv "DUCC0_OPTIMIZATION" "portable-strip"))))))
     (native-inputs
-     (list pybind11
+     (list cmake-minimal
+           pybind11
+           python-nanobind
            python-pytest
-           python-setuptools
-           python-wheel))
+           python-pytest-xdist
+           python-scikit-build-core
+           python-setuptools))
     (propagated-inputs
      (list python-numpy))
     (home-page "https://gitlab.mpcdf.mpg.de/mtr/ducc")
-- 
2.48.1

